Code/Tools/ProjectManager/Source/TextOverflowWidget.cpp

```diff
@@ -0,0 +1,157 @@
+ *
+ */
+
+#include <TextOverflowWidget.h>
+#include <ExternalLinkDialog.h>
+#include <SettingsInterface.h>
+
+#include <QVBoxLayout>
+#include <QScrollArea>
+#include <QRegularExpression>
+#include <QDesktopServices>
+
+namespace O3DE::ProjectManager
+{
+    static QString overflowLink = "OverflowLink";
+    static int maxTextLength = 100;
+
+    // Elides text based on character length with html link tags removed
+    QString ElideLinkedText(const QString& text, int maxLength)
+    {
+        QRegularExpression linksRegex("</?a(|\\s+[^>]+)>");
+        QRegularExpressionMatchIterator matches = linksRegex.globalMatch(text);
+        int displayLength = 0;
+        int truncateEnd = 0;
+        int matchIndex = 0;
+
+        QRegularExpressionMatch match;
+
+        if (matches.hasNext())
+        {
+            int lastMatchLength = 0;
+            while (matches.hasNext())
+            {
+                match = matches.next();
+                int matchStartIndex = match.capturedStart();
+                int matchLength = match.capturedLength();
+
+                // If at the start of a new link tag mark this index
+                // to truncate from here in-case the link text is too long.
+                displayLength += matchStartIndex - truncateEnd - lastMatchLength;
+                truncateEnd = matchStartIndex;
+
+                // Exit because we've run out of displayed characters and truncate to last avaliable character
+                if (displayLength >= maxLength)
+                {
+                    truncateEnd -= displayLength - maxLength;
+                    break;
+                }
+
+                lastMatchLength = matchLength;
+                ++matchIndex;
+            }
+
+            // If there is still room left after we process all matches then
+            // offset by the last match length and the remaining display characters
+            if (displayLength < maxLength)
+            {
+                truncateEnd += lastMatchLength + maxLength - displayLength;
+                displayLength = maxLength;
+            }
+        }
+        else
+        {
+            displayLength = text.length();
+
+            if (displayLength > maxLength)
+            {
+                truncateEnd = maxLength;
+            }
+        }
+
+        // Truncate and add link to full texts
+        if (displayLength >= maxLength)
+        {
+            // Append closing tag if link text got truncated %2
+            return QString("%1%2 <a href=\"%3\">Read More...</a>")
+                .arg(text.leftRef(truncateEnd), match.isValid() && matchIndex % 2 == 1 ? match.captured() : "", overflowLink);
+        }
+
+        return text;
+    }
+
+    TextOverflowDialog::TextOverflowDialog(const QString& title, const QString& text, QWidget* parent)
+        : QDialog(parent)
+    {
+        setWindowTitle(title);
+        setModal(true);
+        setObjectName("textOverflowDialog");
+        setMinimumSize(600, 700);
```

Comment:
700 is right at the limit of max height for project manager, we might want to consider capping at 600 to be safe

Reply:
Lowered to 600